crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -49,6 +49,30 @@ impl Settlement {
         }
 
         let auction = persistence.get_auction(solution.auction_id()).await?;
+        let (winner, winner_score, deadline) =
+            persistence.get_winner(solution.auction_id()).await?;
```

Comment:
Let's give this a domain type (but not expose this type on `Settlement`). I'm not super happy about winner (as this to me is just the solver address). Maybe `Bid`, `Solution` (Solution is already quite loaded in our domain), `SolverPromise` or something similar?

We should also bear in mind that in the future we may have multiple winning bids (in the case of combinatorial auctions).

I think conceptually this is pretty similar to the existing `domain::competition::Solution` type (as technically we also need to check that the promised orders were fulfilled) but it seems we are not fully persisting this yet. We could probably recover it from the competition table though (modulo solution id).

Reply:
> We could probably recover it from the competition table though (modulo solution id)

This requires either:
1. Saving solution id so it can be fetched
2. Making solution id optional field on Solution
3. Making another subtype that wont have solution id, let's say Solution and SolutionWithId.

Reply:
get winner now fetches winning solution from solver competition db. solution id refactored with (2) since (1) would take too long to implement.